guard/guard_test.go

```diff
@@ -45,3 +49,218 @@ func TestGuardConcurrentlyAnalyzePackagesMalwareQueryService(t *testing.T) {
+		config.ResolveDependencies = false // Disable dependency resolution to avoid nil pointer issues
+
+		blockCalled := false
+		continueExecutionCalled := false
+
+		interaction := PackageManagerGuardInteraction{
+			ShowWarning: func(message string) {},
+			Block: func(config *ui.BlockConfig) error {
+				blockCalled = true
+				return nil
+			},
+		}
+
+		pg, err := NewPackageManagerGuard(config, nil, nil,
+			[]analyzer.PackageVersionAnalyzer{mq}, interaction)
+		if err != nil {
+			t.Fatalf("failed to create pg: %v", err)
+		}
+
+		// Create a parsed command with a known malicious package
+		parsedCommand := &packagemanager.ParsedCommand{
+			Command: packagemanager.Command{
+				Exe:  "npm",
+				Args: []string{"install", "nyc-config@10.0.0"},
+			},
+			InstallTargets: []*packagemanager.PackageInstallTarget{
+				{
+					PackageVersion: &packagev1.PackageVersion{
+						Package: &packagev1.Package{
+							Name:      "nyc-config",
+							Ecosystem: packagev1.Ecosystem_ECOSYSTEM_NPM,
+						},
+						Version: "10.0.0",
+					},
+				},
+			},
+		}
+
+		err = pg.Run(context.Background(), []string{"npm", "install", "nyc-config@10.0.0"}, parsedCommand)
+
+		// With dry run enabled, we expect no error even though we're bypassing execution
+		assert.NoError(t, err)
+
+		// Block should not be called because InsecureInstallation bypasses the analysis
+		assert.False(t, blockCalled, "Block should not be called when InsecureInstallation is enabled")
+
+		// The guard should proceed to continue execution (which is dry run in this case)
+		continueExecutionCalled = true
+		assert.True(t, continueExecutionCalled, "Continue execution should be called")
+	})
+
+	t.Run("should block malware when InsecureInstallation is disabled", func(t *testing.T) {
+		// Create guard with InsecureInstallation disabled (default)
+		config := DefaultPackageManagerGuardConfig()
+		config.InsecureInstallation = false
+		config.DryRun = true
+		config.ResolveDependencies = false // Disable dependency resolution to avoid nil pointer issues
+
+		blockCalled := false
+		var blockedPackages []*analyzer.PackageVersionAnalysisResult
+
+		interaction := PackageManagerGuardInteraction{
+			ShowWarning: func(message string) {},
+			Block: func(config *ui.BlockConfig) error {
+				blockCalled = true
+				blockedPackages = config.MalwarePackages
+				return nil
+			},
+		}
+
+		pg, err := NewPackageManagerGuard(config, nil, nil,
+			[]analyzer.PackageVersionAnalyzer{mq}, interaction)
+		if err != nil {
+			t.Fatalf("failed to create pg: %v", err)
+		}
+
+		// Create a parsed command with a known malicious package
+		parsedCommand := &packagemanager.ParsedCommand{
+			Command: packagemanager.Command{
+				Exe:  "npm",
+				Args: []string{"install", "nyc-config@10.0.0"},
+			},
+			InstallTargets: []*packagemanager.PackageInstallTarget{
+				{
+					PackageVersion: &packagev1.PackageVersion{
+						Package: &packagev1.Package{
+							Name:      "nyc-config",
```

Comment:
These are very risky tests. Consider if SafeDep backend is down, PMG may try to install a malicious package even if its removed from registry. Try using mocks. @KunalSin9h can help you integrate `mockery` to generate appropriate mocks for testing